packages/cli/src/commands/rewards/show.ts

```diff
@@ -0,0 +1,293 @@
+import { Validator, ValidatorGroup } from '@celo/contractkit/lib/wrappers/Validators'
+import { eqAddress } from '@celo/utils/lib/address'
+import { concurrentMap } from '@celo/utils/lib/async'
+import { flags } from '@oclif/command'
+import BigNumber from 'bignumber.js'
+import { cli } from 'cli-ux'
+import { EventLog } from 'web3/types'
+import { BaseCommand } from '../../base'
+import { newCheckBuilder } from '../../utils/checks'
+import { Flags } from '../../utils/command'
+
+interface VoterReward {
+  group: string
+  groupName: string
+  totalVoterPayment: BigNumber
+  blockNumber: number
+  activeAddressVotes?: BigNumber
+  activeGroupVotes?: BigNumber
+}
+
+interface ValidatorReward {
+  validator: string
+  validatorName: string
+  validatorPayment: BigNumber
+  validatorScore: BigNumber
+  group: string
+  blockNumber: number
+}
+
+interface ValidatorGroupReward {
+  group: string
+  groupName: string
+  validator: string
+  groupPayment: BigNumber
+  validatorScore: BigNumber
+  blockNumber: number
+}
+
+export default class Show extends BaseCommand {
+  static description =
+    'Show rewards information about a voter, registered Validator, or Validator Group'
+
+  static flags = {
+    ...BaseCommand.flags,
+    voter: Flags.address({ description: 'Voter to show rewards for' }),
+    validator: Flags.address({ description: 'Validator to show rewards for' }),
+    group: Flags.address({ description: 'Validator Group to show rewards for' }),
+    epochs: flags.integer({
+      default: 1,
+      description: 'Show results for the last N epochs',
+    }),
+  }
+
+  static args = []
+
+  static examples = ['show --address 0x5409ed021d9299bf6814279a6a1411a7e866a631']
+
+  async run() {
```

Comment:
In general, i believe this command does too much by itself.. and most of the things here should live in contractKit.

1) **Notion about epoch.** It would be nice to ask ContractKit `getCurrentEpoch` `epochToBlockNumber` etc. Instead of doing the computation by ourselves. "Epoch" is a notion that should be included in contractKit

2) **Notion about rewards** 

Rewards are by epoch, not by block. This command has encoded the notion that to get the rewards you need to query the logs of the last block of the epoch. That notion belongs to contractKit. So you could say `Validator.getValidatorRewards(epochNumber)` and the same for all the others.

In fact, we could have a `Accounts.getRewards(epoch)` or send a epoch range.

Also there's a small gain that we can achieve by saying `give the rewards for this epoch range`. You could cache the validator & validator group details, since they would probably be the same, and you are querying for them on every block.

But as a rule of thumb, if we are doing too much computation here, we should ask ourselves, if this is something that should live in contractKit, and thus be available for any developer as a simple api.

Reply:
Ok, I've added `getCurrentEpoch`, `epochToBlockNumber`, and moved most the reward processing code to contractkit.  

I'm finding the code loses clarity when trying to pipeline multiple epochs, or avoid some redundant queries.  And we actually want the Validator score from each epoch.

So for those reasons I've made the rewards retrieval API single epoch, and not optimized redundant validator group queries.